app/src/main/java/com/nilhcem/blenamebadge/ui/badge_preview/PreviewBadge.kt

```diff
@@ -0,0 +1,141 @@
+package com.nilhcem.blenamebadge.ui.badge_preview
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.graphics.RectF
+import android.graphics.Canvas
+import android.graphics.Paint
+import android.graphics.Color
+import android.graphics.Rect
+import android.graphics.drawable.Drawable
+import android.support.annotation.Nullable
+import android.util.AttributeSet
+import android.view.View
+import com.nilhcem.blenamebadge.R
+
+import java.math.BigInteger
+import java.util.ArrayList
+
+class PreviewBadge : View {
```

Comment:
Why's there a need to implement a custom view?

Reply:
Sir, In order to achieve the best visual of the LED BADGE, I had to implement a custom view. If I relied upon the support libraries, it would not yield the result I was expecting. The only way was to create a Custom View which takes the array of hex's and renders them exactly how they would look on the badge itself. My motive was to give user the best experience with the badge, therefore I chose to mimic the exact hardware into a simple software simulation.

After getting the base ready, It was way easier to display the bitmaps, I just had to pass the list of the hexed Strings. I have the codebase ready for the bitmaps as well, you can watch the results below:
![Screenshot_20190325-211831_Badge Magic](https://user-images.githubusercontent.com/11988517/54967042-3689be00-4f9c-11e9-94d9-ff8fbb1fcf93.jpg):

Here, I just need to change the drawable Id's to get the result. Easy after getting the Custom View Implemented in the core of the project :)

Feel free to change anything you please.
Thanks and I look forward to getting more suggestions from you